File: packages/core-skills/_shared/prompt-loader.ts
/**
 * Shared loader utilities for runtime skills.
 * Phase 2: only the trader-profiles bridge is needed; prompt template
 * interpolation lands with Phase 3 skill scaffolding.
 */

import traderProfilesData from "./trader-profiles.json";

export interface SharedTraderProfile {
  handle: string;
  name: string;
  styleShort: string;
  styleLong: string;
  dimensions: string;
}

export const SHARED_TRADER_PROFILES: SharedTraderProfile[] =
  traderProfilesData as SharedTraderProfile[];

/** Minimal Mustache-style {placeholder} interpolation. Used by Phase 3 skills. */
export function renderPrompt(template: string, vars: Record<string, string>): string {
  return template.replace(/\{(\w+)\}/g, (_, key) =>
    Object.prototype.hasOwnProperty.call(vars, key) ? vars[key] : `{${key}}`
  );
}

File: packages/core-skills/_shared/prompt_loader.py
"""
Shared loader utilities for runtime skills (Python side).

Phase 2: only the trader-profiles bridge is needed; prompt template
interpolation lands with Phase 3 skill scaffolding.
"""
from __future__ import annotations

import json
import os
import re
from typing import Any


_HERE = os.path.dirname(os.path.abspath(__file__))
_PROFILES_PATH = os.path.join(_HERE, "trader-profiles.json")


def load_trader_profiles() -> list[dict[str, Any]]:
    """Load the canonical trader-profiles.json shared with the TS frontend."""
    with open(_PROFILES_PATH, encoding="utf-8") as f:
        return json.load(f)


_PLACEHOLDER_RE = re.compile(r"\{(\w+)\}")


def render_prompt(template: str, **vars: str) -> str:
    """Mustache-style {placeholder} interpolation. Used by Phase 3 skills."""
    def sub(match: re.Match[str]) -> str:
        key = match.group(1)
        return vars[key] if key in vars else match.group(0)
    return _PLACEHOLDER_RE.sub(sub, template)
